tools/lsp-wake/main.cpp

```diff
@@ -641,13 +617,12 @@ int main(int argc, const char **argv) {
     wcl::log::subscribe(std::make_unique<wcl::log::FormatSubscriber>(log_file.rdbuf()));
   }
 
-  if (argc >= 2) {
-    instantiateServer(argv[1]);
-  } else {
-    instantiateServer(wcl::make_canonical(find_execpath() + "/../../share/wake/lib"));
-  }
+  instantiateServerImpl();
+
   // Process requests until something goes wrong
   lspServer->processRequests();
+
+  delete lspServer;
```

Comment:
👀 something something unique_ptr or something something defer?

Reply:
Normally yes, but there isn't a good owner of the unique_ptr. There isn't really an entry point for this code in VSCode

Reply:
is `lspServer` global? Where does it get allocated?

Reply:
Then it can be a global std::unique_ptr

Reply:
:pikachu-face:

Reply:
done :)